src/components/SubNavigation/hooks/useShortenedTitle.ts

```diff
@@ -0,0 +1,89 @@
+import {useCallback, useEffect, useState} from 'react';
+
+const iconWidth = 44; //px
+
+const widthConfig = {
+    // in px
+    burgerButton: iconWidth,
+    shareButton: iconWidth,
+    miniTocIcon: iconWidth,
+    maxPadding: 12,
+    gap: 4,
+};
+
+const widthWithBurger =
+    widthConfig.burgerButton +
+    widthConfig.shareButton +
+    widthConfig.miniTocIcon +
+    2 * widthConfig.maxPadding +
+    3 * widthConfig.gap;
+
+const widthWithoutBurger =
+    widthConfig.shareButton +
+    widthConfig.miniTocIcon +
+    2 * widthConfig.maxPadding +
+    2 * widthConfig.gap;
+
+const speacialChars = new Set([',', '.', '?', '!', '-', '_', ':', ';', '(', '[', '{']);
+
+const getAnotherContentWidth = (hideBurger: boolean) =>
+    hideBurger ? widthWithoutBurger : widthWithBurger;
+// the values are calculated empirically
+const getSymbolSizeQuotient = (hideBurger: boolean) => (hideBurger ? 1 / 7.0 : 1 / 6.5);
+
+// ",...", "?...", "!...", "-...", "_...", "(...", ":..." - looks bad
```

Comment:
what is the result of this hook? why didn't css work here for visual string trimming ?

Reply:
I made a big mistake, and I didn't know that this task could be solved with CSS alone. Now I have used CSS, and now this hook is not needed